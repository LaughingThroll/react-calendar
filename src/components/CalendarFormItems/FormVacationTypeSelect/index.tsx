import React from 'react'
import { FormItem } from '../../common'
import { Select, Option } from '../../common/Select'

import { ControlForSelect } from '../../../types/controls'
import { VacationTypes } from '../../../types/model/vacation'

export interface FormVacationTypeSelectProps {
  select: ControlForSelect
  classNames?: string[]
}

const FormVacationTypeSelect: React.FC<FormVacationTypeSelectProps> = ({ select, classNames = [] }) => {
  return (
    <FormItem title="Vac Type" classNames={classNames}>
      <Select onChange={select.changeValue} value={select.value} name="types">
        <Option title="Paid Day Off (PD)" value={VacationTypes.UN_PAID} />
        <Option title="Paid Day On (PD)" value={VacationTypes.PAID} />
      </Select>
    </FormItem>
  )
}

export default React.memo(FormVacationTypeSelect)
