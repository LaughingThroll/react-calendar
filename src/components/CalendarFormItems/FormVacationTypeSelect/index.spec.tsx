import { shallow } from 'enzyme'
import FormVacationTypeSelect, { FormVacationTypeSelectProps } from './'

const setUp = (props: FormVacationTypeSelectProps) => shallow(<FormVacationTypeSelect {...props} />)

describe('FormVacationTypeSelect render', () => {
  it('default render', () => {
    const props: FormVacationTypeSelectProps = { select: { changeValue: jest.fn(), value: '' } }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
  })
})
