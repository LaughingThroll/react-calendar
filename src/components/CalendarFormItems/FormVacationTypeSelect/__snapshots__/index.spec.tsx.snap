// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormVacationTypeSelect render default render 1`] = `
<Memo(FormItem)
  classNames={Array []}
  title="Vac Type"
>
  <Memo(Select)
    name="types"
    onChange={[MockFunction]}
    value=""
  >
    <Memo(Option)
      title="Paid Day Off (PD)"
      value="UnPaid"
    />
    <Memo(Option)
      title="Paid Day On (PD)"
      value="Paid"
    />
  </Memo(Select)>
</Memo(FormItem)>
`;
